pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -2849,15 +2857,17 @@ func TestUpdatePodOrphanWithNewLabels(t *testing.T) {
 			manager.dsStore.Add(ds1)
 			manager.dsStore.Add(ds2)
 
-			pod := newPod("pod1-", "node-0", simpleDaemonSetLabel, nil)
+			pod := newPod("pod1-", testTenant, metav1.NamespaceDefault, "node-0", simpleDaemonSetLabel, nil)
 			prev := *pod
 			prev.Labels = map[string]string{"foo2": "bar2"}
 			bumpResourceVersion(pod)
 			manager.updatePod(&prev, pod)
 			if got, want := manager.queue.Len(), 2; got != want {
 				t.Fatalf("queue.Len() = %v, want %v", got, want)
 			}
-			if got, want := getQueuedKeys(manager.queue), []string{"default/foo1", "default/foo2"}; !reflect.DeepEqual(got, want) {
+			expectedKey1, _ := cache.DeletionHandlingMetaNamespaceKeyFunc(ds1)
+			expectedKey2, _ := cache.DeletionHandlingMetaNamespaceKeyFunc(ds2)
```

Comment:
Better to use MetaNamespaceKeyFunc instead of DeletionHandlingMetaNamespaceKeyFunc. They are the same under the hood, but we are not dealing with deletiong here.

Reply:
fixed